java-client/src/generated/java/org/opensearch/client/opensearch/indices/AnalyzeRequest.java

```diff
@@ -483,17 +623,16 @@ public AnalyzeRequest build() {
     );
 
     protected static void setupAnalyzeRequestDeserializer(ObjectDeserializer<AnalyzeRequest.Builder> op) {
-
         op.add(Builder::analyzer, JsonpDeserializer.stringDeserializer(), "analyzer");
         op.add(Builder::attributes, JsonpDeserializer.arrayDeserializer(JsonpDeserializer.stringDeserializer()), "attributes");
         op.add(Builder::charFilter, JsonpDeserializer.arrayDeserializer(CharFilter._DESERIALIZER), "char_filter");
         op.add(Builder::explain, JsonpDeserializer.booleanDeserializer(), "explain");
         op.add(Builder::field, JsonpDeserializer.stringDeserializer(), "field");
         op.add(Builder::filter, JsonpDeserializer.arrayDeserializer(TokenFilter._DESERIALIZER), "filter");
+        op.add(Builder::index, JsonpDeserializer.stringDeserializer(), "index");
```

Comment:
The index is not part of the JSON payload, is it? (it is URL path parameter)

Reply:
Correct, we had a few request classes that'd been hand edited to allow deserializing the path params from JSON, so had incorporated it without to avoid breaking changes.  See https://github.com/opensearch-project/opensearch-java/pull/723

The problem we run into is that requests which have required path parameters are impossible to deserialize otherwise due to the required fields.

But can revert it everywhere if you think?

Reply:
> The index is not part of the JSON payload, is it? (it is URL path parameter)

I think the problem here is that this is not the JSON that will be sent to the server but I workaround to restore the state of the object. Yes, I would revert it everywhere, the path / query / etc params should not be part of the request payload (unless they are repeated for some reasons). Thank you.

Reply:
They're only deserializable, they won't be serialized into the JSON sent to the server

Reply:
Exactly, so when they will be serialized into the JSON?

Reply:
I believe that it's being used to construct the request shapes from externally generated JSON. 

If we're going to remove the ability to deserialize the path parameters, then we should completely remove deserialization from the request objects as it's unusable in the majority of cases otherwise. Or provide some other mechanism to pass the path (& query) params separate to the JSON payload for deserialization

Reply:
> Or provide some other mechanism to pass the path (& query) params separate to the JSON payload for deserialization

Yes, I think that would be the right option.